lib/scala/downloader/src/main/java/org/enso/downloader/http/FilesMultipartBodyPublisher.java

```diff
@@ -0,0 +1,48 @@
+package org.enso.downloader.http;
+
+import java.io.IOException;
+import java.net.http.HttpRequest;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This utility class adds support for {@code multipart/form-data} content-type as there is no such
+ * support in {@code java.net.http} JDK.
+ *
+ * <p>Inspired by <a href="https://stackoverflow.com/a/56482187/4816269">SO</a>.
+ */
+public class FilesMultipartBodyPublisher {
+  public static HttpRequest.BodyPublisher ofMimeMultipartData(
```

Comment:
I don't like us implementing such utilities ourselves instead of using existing well-tested solutions.

This works nicely for simple cases, but how do we guarantee that it will work well in general? Should we try testing it as well as an existing solution would? That adds a lot of maintenance burden on us.

> Example 1: the boundary.

I see it is generated as a UUID. I guess it is extremely unlikely that the same UUID would appear inside of the contents of one of the parts submitted as part of this form, but it is not 100% guaranteed.

I'm a bit nitpicky here as indeed it is going to be **very** unlikely to run into this kind of collision. But was this even considered when this code was written?

This is a seemingly simple operation but it may raise many uncertainities like the example above. IMO relying on an existing solution is better as it does not put the maintenance burden of ensuring this solution is bug free, on us.

Reply:
The `downloader` project makes only a very specific kind of HTTP requests and expects a very specific and narrow set of HTTP responses. We are in a full control over these reqs and responses. This project is only used for library uploading and downloading, which is also covered by tests.

Implementing a very simple handling of `multipart/form-data` is a minimal price we have to pay here. Please keep in mind that we are trying very hard, in general, to reduce the number of classes that will eventually get into `runtime.jar`.